src/fennel/specials.fnl

```diff
@@ -1149,12 +1149,21 @@ modules in the compiler environment."
                      "expected each macro to be function" ast)
     (tset scope.macros k v)))
 
-(fn resolve-module-name [{: filename 2 second} _scope _parent opts]
+(fn resolve-module-name [ast _scope _parent opts]
   ;; Compile module path to resolve real module name.  Allows using
   ;; (.. ... :.foo.bar) expressions and self-contained
   ;; statement-expressions in `require`, `include`, `require-macros`,
   ;; and `import-macros`.
-  (let [filename (or filename (and (utils.table? second) second.filename))
+  ;;
+  ;; The first statement in the AST will be an internal function such as
+  ;; `import-macros`, which wont have an appropriate filename to give to the
+  ;; lua chunk.
+  ;;
+  ;; We instead pull the filename from the user-suppied expression. If the user
+  ;; only gave a literal string the filename will be nil, but as that
+  ;; expression is static, that will be inconsequential.
+  (let [{2 second} ast
+        filename (or (and (utils.list? second) second.filename) :unknown)
```

Comment:
Changed from `table?` to `list?` as the argument will always be a list? At least, it never seems to be a plain table.